src/OnchainKitProvider.tsx

```diff
@@ -51,6 +56,47 @@ export function OnchainKitProvider({
     return onchainKitConfig;
   }, [address, apiKey, chain, config, projectId, rpcUrl, schemaId]);
 
+  // Check the React context for WagmiProvider and QueryClientProvider
+  const { providedWagmiConfig, providedQueryClient } =
+    useProviderDependencies();
+
+  const defaultConfig = useMemo(() => {
+    // IMPORTANT: Don't create a new Wagmi configuration if one already exists
+    // This prevents the user-provided WagmiConfig from being overriden
+    return (
+      providedWagmiConfig ||
+      createWagmiConfig({
+        apiKey,
+        appName: value.config.appearance.name,
+        appLogoUrl: value.config.appearance.logo,
+      })
+    );
+  }, [
+    apiKey,
+    providedWagmiConfig,
+    value.config.appearance.name,
+    value.config.appearance.logo,
+  ]);
+  const defaultQueryClient = useMemo(() => {
+    // IMPORTANT: Don't create a new QueryClient if one already exists
+    // This prevents the user-provided QueryClient from being overriden
+    return providedQueryClient || new QueryClient();
+  }, [providedQueryClient]);
+
+  // If both dependencies are missing, return a context with default parent providers
+  // If only one dependency is provided, expect the user to also provide the missing one
+  if (!providedWagmiConfig && !providedQueryClient) {
+    return (
+      <WagmiProvider config={defaultConfig}>
+        <QueryClientProvider client={defaultQueryClient}>
+          <OnchainKitContext.Provider value={value}>
+            {children}
+          </OnchainKitContext.Provider>
+        </QueryClientProvider>
+      </WagmiProvider>
+    );
```

Comment:
since Wagmi requires QueryClient, the only time 1 dep is missing should be when there is a QueryClient, but no Wagmi?   Seems like an edge case, but would be possible to support this usecase, no?  or, should we adjust the console.errors at least to only state using default if both don't exist and otherwise prompt the user to fix?

Reply:
if 1 dep is missing, we shouldn't modify the default behavior and allow the app to display the typical error message instead

<img width="1208" alt="image" src="https://github.com/user-attachments/assets/9837ddd8-53f4-41dc-9c0b-853b56ce1492">

Reply:
in my mind, if you provide 1 of the 2 deps then it should be the same logic as what it is currently (before this PR) which is `Unhandled Runtime Error`